cpp/src/arrow_chunk_writer.cc

```diff
@@ -90,7 +90,42 @@ Result<std::shared_ptr<arrow::Table>> ExecutePlanAndCollectAsTable(
 
 // implementations for VertexPropertyChunkWriter
 
-Status VertexPropertyWriter::Validate(
+Status VertexPropertyWriter::validate(const IdType& count,
+                                      ValidateLevel validate_level) const
+    noexcept {
+  // use the writer's validate level
+  if (validate_level == ValidateLevel::default_validate)
+    validate_level = validate_level_;
+  // no validate
+  if (validate_level == ValidateLevel::no_validate)
+    return Status::OK();
+  // weak & strong validate
+  if (count < 0)
```

Comment:
please follow the [google c++ code style](https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching), if body can't not fit on one line, use braces